streams/delimiter_stream.ts

```diff
@@ -179,7 +183,7 @@ export class DelimiterStream extends TransformStream<Uint8Array, Uint8Array> {
     this.#matchIndex = matchIndex;
     if (chunkStart === 0) {
       bufs.push(chunk);
-    } else if (chunkStart !== length - 1) {
+    } else if (chunkStart < length) {
```

Comment:
**comment:** This is the fix for the first issue: This is an off-by-one-error here.